cross compiling for QNX -> CDR #define HAVE_CXX0X not set [8999]

When cross compiling for QNX cdr builds fine and HAVE_CXX0X gets set, but when fastrtps starts to build it fails with the following error:

/home/asobhy/workspace/ros2_foxy/src/eProsima/Fast-DDS/src/cpp/dynamic-types/AnnotationParameterValue.cpp:1696:13:   required from here
/home/asobhy/workspace/ros2_foxy/install/x86_64/include/fastcdr/Cdr.h:429:32: error: 'const struct std::array<unsigned char, 4>' has no member named 'serialize'
                         type_t.serialize(*this);


conplains about the array template because in CDR.h  HAVE_CXX0X is not set and the array templates are not included for serialize and deserialize overloaded operator<< and operator>>

SOLUTION:
-replace in fartrtps in https://github.com/eProsima/Fast-DDS/blob/master/cmake/common/check_configuration.cmake, line 18 with the following:

if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR
        CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR
        CMAKE_CXX_COMPILER_ID MATCHES "QCC")

FastRTPS:
commit 40568fa4784e846f95c461608d43a2b57eaef55b (HEAD -> 2.0.x, tag: v2.0.1, origin/2.0.x)

--- a/cmake/common/check_configuration.cmake
+++ b/cmake/common/check_configuration.cmake
@@ -16,7 +16,7 @@ macro(check_stdcxx)
     # Check C++11
     include(CheckCXXCompilerFlag)
     if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR
-        CMAKE_CXX_COMPILER_ID MATCHES "Clang"OR
+        CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR
         CMAKE_CXX_COMPILER_ID MATCHES "QCC")
         check_cxx_compiler_flag(-std=c++14 SUPPORTS_CXX14)
         set(HAVE_CXX14 0)

--- a/cmake/common/check_configuration.cmake
+++ b/cmake/common/check_configuration.cmake
@@ -16,7 +16,8 @@ macro(check_stdcxx)
     # Check C++11
     include(CheckCXXCompilerFlag)
     if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR
-        CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        CMAKE_CXX_COMPILER_ID MATCHES "Clang"OR
+        CMAKE_CXX_COMPILER_ID MATCHES "QCC")
         check_cxx_compiler_flag(-std=c++14 SUPPORTS_CXX14)
         set(HAVE_CXX14 0)
         set(HAVE_CXX1Y 0)

--- a/cmake/modules/FindThirdpartyBoost.cmake
+++ b/cmake/modules/FindThirdpartyBoost.cmake
@@ -1,19 +1,19 @@
 # Test compilation of boost on thirdparty directory
-# and set 
+# and set
 
 # (This file is almost an identical copy of the original FindGTest.cmake file,
 #  feel free to use it as it is or modify it for your own needs.)
 
-set(THIRDPARTY_BOOST_INCLUDE_DIR 
-    ${PROJECT_SOURCE_DIR}/thirdparty/boost/include 
-    CACHE 
+set(THIRDPARTY_BOOST_INCLUDE_DIR
+    ${PROJECT_SOURCE_DIR}/thirdparty/boost/include
+    CACHE
     FILEPATH
     "Path to thirdparty/boost"
 )
 
 find_package(Threads REQUIRED)
 
-if(WIN32 OR APPLE)
+if(WIN32 OR APPLE OR QNX)
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT})
 else() # Posix
     set(THIRDPARTY_BOOST_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT} rt)

